### PR TITLE
Add tests, update todo

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

File: TODO.md [hunk 1]
```diff
@@ -1,34 +1,44 @@
  # TODO
 - Design
-    - [ ] Input: bypass source exceptions in `Input.fill`
-    - [ ] Input: `eof` vs `exhausted`
-    - [ ] `Float.reverseByteOrder`
-    - [x] Output: `flush` size contract
-    - [ ] case: `Bytes` -> `[Input]`. When we should close `Bytes`?
+    - Input: bypass source exceptions in `Input.fill`
+    - Input: `eof` vs `exhausted`
+    - `Float.reverseByteOrder`
```
Comment (qwwdfsad) [inline TODO.md]: Already fixed

File: core/commonTest/src/kotlinx/io/tests/BytesTest.kt [hunk 2]
```diff
@@ -52,6 +52,178 @@ class BytesTest {
...
+            writeInt(0x12345678)
+            writeDouble(1.25)
+            writeFloat(1.25f)
+            writeLong(0x123456789abcdef0)
+
+            writeUTF8String("OK\n")
+            val text = listOf(1, 2, 3).joinToString(separator = "|")
+            writeUTF8String("$text\n")
+        }.useInput {
+            //        assertEquals(9999 + 1 + 2 + 4 + 8 + 4 + 8 + 3 + 5, p.remaining)
```
Comment (qwwdfsad) [inline core/commonTest/src/kotlinx/io/tests/BytesTest.kt]: commented code (here and below) should be either uncommented or removed

File: core/commonTest/src/kotlinx/io/tests/InputPrimitivesTest.kt [hunk 3]
```diff
@@ -94,4 +97,65 @@ class InputPrimitivesTest {
         assertTrue(eof(), "EOF")
         assertFails { readUByte() }
     }
+
+    @Test
+    fun smokeTest() {
```
Comment (qwwdfsad) [inline core/commonTest/src/kotlinx/io/tests/InputPrimitivesTest.kt]: Missing `test` prefix

File: core/commonTest/src/kotlinx/io/tests/BytesTest.kt [hunk 4]
```diff
@@ -52,6 +52,178 @@ class BytesTest {
...
 */
 
-        //assertTrue { bytes.isEmpty }
+        //assertTrue { bytes.eof() }
     }
-}
\ No newline at end of file
+
+    @Test
+    fun smokeMultiBufferTest() {
```
Comment (qwwdfsad) [inline core/commonTest/src/kotlinx/io/tests/BytesTest.kt]: missing test prefix

**Comment by e5l** [comment]
Merged manually.

**Pull request closed without merging**